l10n_be_cooperator/migrations/16.0.3.0.0/pre-legal_form.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-FileCopyrightText: 2025 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+def migrate(cr, version):
+
+    # temp column to keep the values of legal_form after we removed it
+    cr.execute(
+        """
+        ALTER TABLE res_partner
+        ADD temp_legal_form VARCHAR;
+        """
+    )
+    # fill the temp_column
+    cr.execute(
+        """
+        UPDATE res_partner
+        SET temp_legal_form = legal_form
+        """
+    )
+    # Remove the old values in res_partner.legal_form to prepare the database for
+    # those fields to be deleted. Otherwise the values added by add_selection
+    # will not be valid anymore but would still be present in the db.
+    cr.execute(
+        """
+        UPDATE res_partner
+        SET legal_form = NULL
+        """
+    )
+
```

Comment:
is this needed? the column will be deleted no matter what. same remark for `subscription_request.company_type`.